cfme/fixtures/v2v.py

```diff
@@ -41,6 +37,292 @@ def conversion_tags(appliance, host_creds):
+    collection = second_provider.appliance.provider_based_collection(second_provider)
+    vm_obj = collection.instantiate(random_vm_name('v2v'),
+                                    second_provider,
+                                    template_name=template(second_provider)['name'])
+
+    request.addfinalizer(lambda: vm_obj.cleanup_on_provider())
+    vm_obj.create_on_provider(timeout=2400, find_in_cfme=True, allow_skip="default",
+         datastore=source_datastore)
+    return vm_obj
+
+
+def _form_data_cluster_mapping(second_provider, provider):
+    # since we have only one cluster on providers
+    source_cluster = second_provider.data.get('clusters', [False])[0]
+    target_cluster = provider.data.get('clusters', [False])[0]
+
+    if not source_cluster or not target_cluster:
+        pytest.skip("No data for source or target cluster in providers.")
+
+    return {
+        'sources': [partial_match(source_cluster)],
+        'target': [partial_match(target_cluster)]
+    }
+
+
+def _form_data_datastore_mapping(second_provider, provider, source_type, target_type):
+    source_datastores_list = second_provider.data.get('datastores', [])
+    target_datastores_list = provider.data.get('datastores', [])
+
+    # assuming, we just have 1 datastore of each type
+    source_datastore = [d.name for d in source_datastores_list if d.type == source_type]
+    target_datastore = [d.name for d in target_datastores_list if d.type == target_type]
+    if (not source_datastores_list or not target_datastores_list or
+            not source_datastore or not target_datastore):
+        pytest.skip("No data for source or target datastore in providers.")
+
+    return {
+        'sources': [partial_match(source_datastore[0])],
+        'target': [partial_match(target_datastore[0])]
+    }
+
+
+def _form_data_network_mapping(second_provider, provider, source_network_name, target_network_name):
+    source_vlans_list = second_provider.data.get('vlans', [])
+    target_vlans_list = provider.data.get('vlans', [])
+
+    # assuming there will be only 1 network matching given name
+    source_network = [v for v in source_vlans_list if v == source_network_name]
+    target_network = [v for v in target_vlans_list if v == target_network_name]
+
+    if not source_vlans_list or not target_vlans_list or not source_network or not target_network:
+        pytest.skip("No data for source or target cluster in providers.")
+
+    return {
+        'sources': [partial_match(source_network[0])],
+        'target': [partial_match(target_network[0])]
+    }
+
+
+@pytest.fixture(scope='function')
+def form_data_single_datastore(request, appliance, second_provider, provider):
+    """Return Infra Mapping form data and vm_obj, encapsulated in list,
+       deployed on correct Datastore for migration."""
+    cluster = provider.data.get('clusters', [False])[0]
+    if not cluster:
+        pytest.skip("No data for cluster available on provider.")
+
+    form_data = (
```

Comment:
Why is this wrapped in parenthesis? should be able to define dict on multi-line without it:

```
form_data = {
    'general': {}
}